### PR TITLE
Inline single use constructor

### PR DESCRIPTION
**Pull request by elharo**
## Description
Standardize on "row number" as the term for location of rows within a group instead of "row position" and introduce other local variables as needed for clarity.

## Motivation and Context
Clarify and simplify code while searching for bug. In particular, standardize on "row number" as the term for location of rows within a group instead of "row position". Previously these terms were used interchangeably in different classes. I found this a tad confusing, especially given that we also have "row IDs" which are not the same thing.

## Impact
None

## Test Plan
CI

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (sdruzkin) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java:73]: Do you mind converting it to IntOpenHashSet or boolean[]?

**Review by sdruzkin — APPROVED** [review]

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java:73]: sounds like a good idea. However, if I do that in this PR changes will slip into presto-iceberg and your approval alone is no longer sufficient, so perhaps that should be a separate change?

Comment (sdruzkin) [reply presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java:73]: You can convert List<Boolean> to the new type in the constructor and keep the interface untouched.

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (sdruzkin) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java:103]: Please add a size check that isRowNumbers.len >= columns.size()

**Review by sdruzkin — APPROVED** [review]

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java]: done

**Review by sdruzkin — DISMISSED** [review]
LGTM

**Comment by elharo** [comment]
Bleah. This does need global committer approval since it's in the ORC parts of presto-hive instead of presto-orc.

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (tdcmeehan) [inline presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java:89]:
```suggestion
    // TODO: make HiveColumnHandle contain whether it's a row number or not. Alternately, define a class
    // that includes both a column handle and the row number boolean.
```
Please remove extraneous commentary so that it's very clear what needs to be done.

**Review by tdcmeehan — COMMENTED** [review]
LGTM, just shorten the TODO.

**Review by arhimondr — DISMISSED** [review]

File: presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java
Comment (elharo) [reply presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java]: done

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**
